src/Evolution/Initialization/AddToDataBox.hpp

```diff
@@ -0,0 +1,145 @@
+          << simple_tag);
+  }
+}
+
+template <
+    typename AddingAction, typename SimpleTag, MergePolicy Policy,
+    typename DbTagsList,
+    Requires<(Policy == MergePolicy::Error or Policy == MergePolicy::Ignore) and
+             not tt::has_inequivalence_v<db::item_type<SimpleTag>>> = nullptr>
+void merge_simple_tag_value(
+    const gsl::not_null<db::DataBox<DbTagsList>*> /*box*/,
+    db::item_type<SimpleTag>&& /*simple_tag*/) noexcept {
+  static_assert(Policy != MergePolicy::Error,
+                "The tag being added does not have an equivalence operator and "
+                "is already in the DataBox. See the first template parameter "
+                "of the 'merge_simple_tag_value' function in the error message "
+                "for the action trying to re-add the simple tag and the second "
+                "template parameter for the simple tag being added.");
+}
+
+template <typename AddingAction, typename ComputeTagsList, MergePolicy Policy,
+          typename... SimpleTags, typename DbTagsList,
+          typename... SimpleTagsToAdd, typename... SimpleTagsToCheck>
+auto merge_into_databox_impl(
+    db::DataBox<DbTagsList>&& box,
+    tuples::TaggedTuple<SimpleTags...>&& simple_tags,
+    tmpl::list<SimpleTagsToAdd...> /*meta*/,
+    tmpl::list<SimpleTagsToCheck...> /*meta*/) noexcept {
+  EXPAND_PACK_LEFT_TO_RIGHT(
+      merge_simple_tag_value<AddingAction, SimpleTagsToCheck, Policy>(
+          make_not_null(&box),
+          std::move(tuples::get<SimpleTagsToCheck>(simple_tags))));
+
+  // Only add compute tags that are not in the DataBox.
+  using compute_tags_to_add = tmpl::remove_if<
+      ComputeTagsList,
+      tmpl::bind<tmpl::list_contains,
+                 tmpl::pin<typename db::DataBox<DbTagsList>::compute_item_tags>,
+                 tmpl::_1>>;
+
+  return db::create_from<db::RemoveTags<>,
+                         db::AddSimpleTags<SimpleTagsToAdd...>,
+                         compute_tags_to_add>(
+      std::move(box), std::move(tuples::get<SimpleTagsToAdd>(simple_tags))...);
+}
+}  // namespace detail
+
+/*!
+ * \ingroup InitializationGroup
+ * \brief Add tags that are not yet in the DataBox.
+ *
+ * How duplicate tags are handled depends on the `MergePolicy` passed. The
+ * default merge policy is to error if the simple tag being added does not have
+ * an equivalence operator and is being added again. `MergePolicy::Ignore` means
+ * that nothing is done if the simple tag being added does not have an
+ * equivalence operator. Tags that have an equivalence operator are check that
+ * the value being added and that is already in the DataBox are the same. If
+ * they aren't, an error occurs. Finally, `MergePolicy::Overwrite` means the
+ * current simple tag in the DataBox is overwritten with the new value passed
+ * in.
+ *
+ * Compute tags that are not in the DataBox are added, ones that are in the
+ * DataBox already are ignored.
+ */
+template <typename AddingAction, typename ComputeTagsList,
```

Comment:
What I meant in the other comment is to default `ComputeTagsList` to `tmp::list<>` here. Is that also what you meant, that you intentionally didn't do it? If so, why? Would be good to keep this as close to `db::create_from` as possible

Reply:
:man_shrugging: I'm leaving it.